File: Assets/js/LanguageButton.js
var labelText : Text;
var languageDescription : String;
var languageCode : String;

function HandleOnClick() {
	VerseManager.SwitchLanguage(languageCode);
	Application.LoadLevel("versesets");
}

function SetCodeAndLanguage(code : String, language : String) {
	languageCode = code;
	languageDescription = language;
	SetLabel(languageDescription);
}

function SetLabel(label : String) {
	labelText.text = label;
}

function SetParent(prt : RectTransform) {
	var rt : RectTransform = GetComponent(RectTransform);
	var oldAPosition = rt.anchoredPosition;
	
	var oldScale = rt.localScale;
	var oldMin = rt.offsetMin;
	var oldMax = rt.offsetMax;
	
//	Debug.Log("offset min = " + oldMin);
//	Debug.Log("offset max = " + oldMax);
	
	rt.SetParent(prt);
	
	rt.anchoredPosition = oldAPosition;
	rt.localScale = oldScale;	
	
//	rt.offsetMin = oldMin;
//rt.offsetMax = oldMax;

	rt.offsetMin.x = 0;
	rt.offsetMax.x = -60;
	
}

function Awake() {
	SetLabel(languageDescription);
}

function Start() {
}

function Update() {
}

File: Assets/cs/LanguageButton.cs
using UnityEngine;
using System;
using UnityEngine.UI;
using UnityEngine.SceneManagement;


public class LanguageButton:MonoBehaviour{
	
	public Text labelText;
	public string languageDescription;
	public string languageCode;
	
	public void HandleOnClick() {
		Action<string> onFinish = delegate( string language) {
			VerseSceneManager.loadVersesets ();
		};
		VerseManager.GetInstance().SwitchLanguage(languageCode, onFinish);
	}
	
	public void SetCodeAndLanguage(string code,string language) {
		languageCode = code;
		languageDescription = language;
		SetLabel(languageDescription);
	}
	
	public void SetLabel(string label) {
		labelText.text = label;
	}
	
	public void SetParent(RectTransform prt) {
		RectTransform rt = GetComponent<RectTransform>();
		Vector2 oldAPosition = rt.anchoredPosition;
		
		Vector3 oldScale = rt.localScale;
		//Vector2 oldMin = rt.offsetMin;
		//Vector2 oldMax = rt.offsetMax;
		
	//	Debug.Log("offset min = " + oldMin);
	//	Debug.Log("offset max = " + oldMax);
		
		rt.SetParent(prt);
		
		rt.anchoredPosition = oldAPosition;
		rt.localScale = oldScale;	
		
	//	rt.offsetMin = oldMin;
	//rt.offsetMax = oldMax;
	
		var tmp_cs1 = rt.offsetMin;
        tmp_cs1.x = 0.0f;
        rt.offsetMin = tmp_cs1;
		var tmp_cs2 = rt.offsetMax;
        tmp_cs2.x = -60.0f;
        rt.offsetMax = tmp_cs2;
		
	}
	
	public void Awake() {
		SetLabel(languageDescription);
	}
	
	public void Start() {
	}
	
	public void Update() {
	}

}
